### PR TITLE
update to gradle 4.8

### PR DESCRIPTION
**Pull request by lbergelson**
* Upgrading our version of gradle and our test dependencies.
  This is a move towards supporting building under java 9/10 although
  some issues remain.

  upgrading from gradle 3.2.1 -> 4.8
  upgrading shadow plugin  1.2.3 -> 2.0.4
  updating scalatest plugin 0.15 -> 0.22

  testng 6.13.1 -> 6.14.3
  scala-library 2.12.1 -> 2.12.6
  scalatest 3.0.1 -> 3.0.5
  pegdown 1.4.2 -> 1.6.0

  updating the gradle wrapper task definition because the way we were doing it has been deprecated
  removing jacoco version specification so we stay up to date automatically

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,19 +11,14 @@ plugins {
     id 'signing'
     id 'jacoco'
     id 'com.palantir.git-version' version '0.5.1'
```
Comment (magicDGS) [inline build.gradle]: Cannot you update also the git-version?

Comment (lbergelson) [reply build.gradle]: That's a good point.  I had avoided it because I didn't want any version name surprises, but it's worthing looking at.

Comment (lbergelson) [reply build.gradle]: It seems fine, I've updated it.

**Comment by lbergelson** [comment]
@cmnbroad Yeah, they seem to systematically rewrite their wrapper script with every release.  It's not clear why since they all seem to work exactly the same.

**Comment by lbergelson** [comment]
They released a 4.8.1 since I pushed this, so I'm bumping to that.  Merging when tests are green.

**Pull request merged**